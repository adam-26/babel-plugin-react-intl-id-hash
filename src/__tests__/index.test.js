// @flow
import path from 'path'
import slash from 'slash'
import pluginTester from 'babel-plugin-tester'
import plugin from '../'

const filename = path.resolve(
  slash(__dirname),
  '..',
  '__fixtures__',
  'messages.js'
)

const standardTest = {
  title: 'basic',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    id: 'App.Components.Greeting.hello',
    defaultMessage: 'id',
    description: 'describe text for translation'
  }
})
`,
}

const missingIdTest = {
  title: 'basic',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    defaultMessage: 'missing id',
    description: 'should not fail when id is missing'
  }
})
`,
}

function pTest(opts: Object) {
  pluginTester(
    Object.assign(
      {
        plugin,
        snapshot: true,
        babelOptions: { filename },
      },
      opts
    )
  )
}

pTest({
  title: 'default',
  tests: [standardTest, missingIdTest],
  pluginOptions: {},
})

pTest({
  title: 'default',
  tests: [standardTest, missingIdTest],
  pluginOptions: { idHash: 'murmur3' },
})
