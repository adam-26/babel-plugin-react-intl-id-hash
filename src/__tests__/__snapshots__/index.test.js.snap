// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    id: 'App.Components.Greeting.hello',
    defaultMessage: 'id',
    description: 'describe text for translation'
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  new: {
    'id': 'GSplhw==',

    defaultMessage: 'id',
    description: 'describe text for translation'
  }
});
"
`;

exports[`basic 2`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    id: 'App.Components.Greeting.hello',
    defaultMessage: 'id',
    description: 'describe text for translation'
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  new: {
    'id': 'GSplhw==',

    defaultMessage: 'id',
    description: 'describe text for translation'
  }
});
"
`;

exports[`missingId 1`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    defaultMessage: 'missing id',
    description: 'should not fail when id is missing'
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  new: {
    defaultMessage: 'missing id',
    description: 'should not fail when id is missing'
  }
});
"
`;

exports[`missingId 2`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    defaultMessage: 'missing id',
    description: 'should not fail when id is missing'
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  new: {
    defaultMessage: 'missing id',
    description: 'should not fail when id is missing'
  }
});
"
`;

exports[`stringId 1`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    \\"id\\": \\"App.Components.Greeting.hello\\",
    \\"defaultMessage\\": \\"id\\",
    \\"description\\": \\"describe text for translation\\"
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  new: {
    \\"id\\": \\"GSplhw==\\",

    \\"defaultMessage\\": \\"id\\",
    \\"description\\": \\"describe text for translation\\"
  }
});
"
`;

exports[`stringId 2`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  new: {
    \\"id\\": \\"App.Components.Greeting.hello\\",
    \\"defaultMessage\\": \\"id\\",
    \\"description\\": \\"describe text for translation\\"
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  new: {
    \\"id\\": \\"GSplhw==\\",

    \\"defaultMessage\\": \\"id\\",
    \\"description\\": \\"describe text for translation\\"
  }
});
"
`;
